components/sup/src/manager/service/mod.rs

```diff
@@ -68,22 +71,23 @@ lazy_static! {
 #[derive(Debug, Serialize)]
 pub struct Service {
     pub service_group: ServiceGroup,
-    pub config: ServiceConfig,
     pub depot_url: String,
     pub spec_file: PathBuf,
     pub spec_ident: PackageIdent,
     pub start_style: StartStyle,
     pub topology: Topology,
     pub update_strategy: UpdateStrategy,
+    pub cfg: Cfg,
+    pub pkg: Pkg,
+    pub sys: Arc<Sys>,
```

Comment:
Why is Sys behind an Arc? Shouldn't it be immutable after start up? Or is this just to save memory instead of cloning?